lstchain/image/modifier.py

```diff
@@ -54,32 +54,51 @@ def add_noise_in_pixels(rng, image, extra_noise_in_dim_pixels,
     return image
 
 
-def smear_light_in_pixels(image, camera_geom, smeared_light_fraction):
+def smear_light_in_pixels(rng, image, camera_geom, smeared_light_fraction):
     """
 
     Parameters
     ----------
+    rng: numpy.random.default_rng  random number generator
+
     image: charges (p.e.) in the camera
 
     camera_geom: camera geometry
 
     smeared_light_fraction: fraction of the light in a pixel that will be
-    distributed equally among its immediate surroundings, i.e. immediate
-    neighboring pixels. Some light is lost for pixels which are at the
-    camera edge and hence don't have all possible neighbors
+    distributed among its immediate surroundings, i.e. immediate
+    neighboring pixels, according to Poisson statistics. Some light is
+    lost for pixels which are at the camera edge and hence don't have all
+    possible neighbors
 
     Returns
     -------
     Modified (smeared) image
 
     """
 
-    # Move a fraction of the light in each pixel (fraction) into its neighbors,
-    # to simulate a worse PSF:
-    q_smeared = (image * camera_geom.neighbor_matrix *
-                 smeared_light_fraction / N_PIXEL_NEIGHBORS)
+    # How many p.e. to smear?  Poisson of the smeared light fraction (clipped to a minimum of 0):
+    pe_to_smear = rng.poisson(np.clip(image * smeared_light_fraction, 0, np.inf))
+
+    # How to distribute the smeared charge among neighboring pixels (multinomial):
+    smeared_charges = np.zeros(shape=(len(image), N_PIXEL_NEIGHBORS))
+    for q in np.unique(pe_to_smear):
+        # generate the p.e. smearing patterns for all pixels with q  p.e.'s to be smeared:
+        num_pixels = np.sum(pe_to_smear == q)
+        # in this way we speed things up a lot through vectorization (vs. doing it pixel-wise):
+        smeared_charges[pe_to_smear == q, :] = rng.multinomial(q, N_PIXEL_NEIGHBORS * [1 / N_PIXEL_NEIGHBORS],
```

Comment:
`np.full(N_PIXEL_NEIGHBORS, 1 / N_PIXEL_NEIGHBORS)` end this can be a global constant.